### PR TITLE
[Elastic Agent] Fix issues with the Gateway     

### PR DESCRIPTION
**Pull request by ph**
## What does this PR do?

This PRs fixes a few issues:
    
1. Fix the flaky tests found in the Gateway (#17709)
2. Fix isses on shutdown (#17760):
    - when the gateway was stuck in a retry loop.
    - when mutliple signal (ctrl-c) were sent to the application.

## Why is it important?

Well it fixes panics.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: x-pack/elastic-agent/pkg/agent/application/fleet_gateway.go [hunk 1]
```diff
@@ -98,6 +98,9 @@ func newFleetGatewayWithScheduler(
 	r fleetReporter,
 	acker fleetAcker,
 ) (*fleetGateway, error) {
+
+	// Backoff implementation doesn't support the using context as the shutdown mechanism.
+	// So we keep a done channel that will be closed when the current context is shutdown.
 	done := make(chan struct{})
```
Comment (michalpristas) [inline x-pack/elastic-agent/pkg/agent/application/fleet_gateway.go:104]: do we need it still?

Comment (ph) [reply x-pack/elastic-agent/pkg/agent/application/fleet_gateway.go:104]: Yeah, backoff doesn't support passing a context to it. so to interrupt the backoff we need to rely on a channel.

Comment (ph) [reply x-pack/elastic-agent/pkg/agent/application/fleet_gateway.go:121]: I've created https://github.com/elastic/beats/issues/18140 in the platform team.

**Comment by ph** [comment]
Doing another set of tests I want to make sure it's solid.
I've run the test in loop since yesterday, I haven't had failures.

**Comment by ph** [comment]
Jenkins test this please

**Comment by ph** [comment]
rebase again..

**Pull request merged**